src/main/java/org/jabref/logic/cleanup/PageFieldCleanup.java

```diff
@@ -0,0 +1,34 @@
+package org.jabref.logic.cleanup;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.model.FieldChange;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+
+/**
+ * adds the article ID of a journal as the page count, but only if the page field is empty
+ */
+public class PageFieldCleanup implements CleanupJob {
+
+    @Override
+    public List<FieldChange> cleanup(BibEntry entry) {
+        List<FieldChange> changes = new ArrayList<>();
+        Optional<String> doiAsString = entry.getField(StandardField.DOI);
+
+        if (doiAsString.isPresent() && !entry.hasField(StandardField.PAGES)) {
+            String articleId = new String();
+            int index = doiAsString.get().length() - 1;
```

Comment:
I would simply use a substring with lastIndexOf(.) to the the part after the last dot.
https://docs.oracle.com/en/java/javase/14/docs/api/java.base/java/lang/String.html#lastIndexOf(java.lang.String)